conans/client/manager.py

```diff
@@ -210,16 +213,50 @@ def info(self, reference, current_path, remote=None, options=None, settings=None
                                               info, registry, graph_updates_info,
                                               remote)
 
+    def _read_profile(self, profile_name):
+        if profile_name:
+            try:
+                profile = self._client_cache.load_profile(profile_name)
+                return profile
+            except Exception:
+                current_profiles = ", ".join(self._client_cache.current_profiles()) or "[]"
+                raise ConanException("Specified profile '%s' doesn't exist.\nExisting profiles: "
+                                     "%s" % (profile_name, current_profiles))
+        return None
+
+    def _mix_settings_and_profile(self, settings, profile_name):
+        '''Mix the specified settings with the specified profile.
+        Specified settings are prioritized to profile'''
+        profile = self._read_profile(profile_name)
```

Comment:
Reading again and again the profile from disk, seems it should be read once, then used multiple times